Description: Patching PHP 5.3.3 to build with GCC 4.5 and DSO libs
Author: Clint Byrum <clint@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/php5/+bug/676672

--- a/ext/ldap/config.m4
+++ b/ext/ldap/config.m4
@@ -50,7 +50,7 @@ AC_DEFUN([PHP_LDAP_SASL_CHECKS], [
     SASL_LIB="-L$LDAP_SASL_LIBDIR -lsasl2"
   fi
   
-  PHP_CHECK_LIBRARY(ldap, sasl_version,
+  PHP_CHECK_LIBRARY(sasl2, sasl_version,
   [
     PHP_ADD_INCLUDE($LDAP_SASL_INCDIR)
     PHP_ADD_LIBRARY_WITH_PATH(sasl2, $LDAP_SASL_LIBDIR, LDAP_SHARED_LIBADD)
--- a/ext/openssl/config0.m4
+++ b/ext/openssl/config0.m4
@@ -17,6 +17,7 @@ if test "$PHP_OPENSSL" != "no"; then
   fi
 
   AC_CHECK_LIB(ssl, DSA_get_default_method, AC_DEFINE(HAVE_DSA_DEFAULT_METHOD, 1, [OpenSSL 0.9.7 or later]))
+  AC_CHECK_LIB(crypto, X509_free, AC_DEFINE(HAVE_DSA_DEFAULT_METHOD, 1, [OpenSSL 0.9.7 or later]))
 
   PHP_SETUP_OPENSSL(OPENSSL_SHARED_LIBADD, 
   [
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -2395,6 +2395,7 @@ AC_DEFUN([PHP_SETUP_OPENSSL],[
     ])
     LIBS=$old_LIBS
     PHP_ADD_LIBRARY(ssl,,$1)
+    PHP_ADD_LIBRARY(crypto,,$1)
 
     PHP_ADD_LIBPATH($OPENSSL_LIBDIR, $1)
   fi
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -2397,6 +2397,7 @@ AC_DEFUN([PHP_SETUP_OPENSSL],[
     ])
     LIBS=$old_LIBS
     PHP_ADD_LIBRARY(ssl,,$1)
+    PHP_ADD_LIBRARY(crypto,,$1)
 
     PHP_ADD_LIBPATH($OPENSSL_LIBDIR, $1)
   fi
